File: back/src/Entity/Campaign.ts
import { Entity, PrimaryGeneratedColumn, Column, ManyToMany, JoinTable, OneToMany } from "typeorm";
import Subscription from "./Subscription";
import Winner from "./Winner";

@Entity()
export default class Campaign {
    constructor() {
        this.name = "";
    }
    @PrimaryGeneratedColumn()
    id?: number;

    @Column()
    title: string;

    @Column({ type: 'datetime' })
    startAt: Date;

    @Column({ type: 'datetime' })
    endsAt: Date;

    @ManyToMany((type) => Subscription, { cascade: false })
    @JoinTable()
    subscription?: Subscription[];

    @ManyToMany((type) => Winner, { cascade: false })
    @JoinTable()
    winner?: Winner[];
}

File: back/src/main/java/com/superlucky/model/Campaign.java
package com.superlucky.model;

import java.util.Date;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;

import org.springframework.data.annotation.Id;

import lombok.Data;

@Entity
@Data
public class Campaign {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public Long id;
    public String title;
    public Date startsAt;
    public Date endsAt;
    public Date sortitionAt;
    public boolean isActive;
    //@OneToMany
    //public List<Winner> winners;
    //@OneToMany
    //public List<Subscription> subscriptions;
    //@OneToMany
    //public Dictionary<Long,Prize> prizes;
}